leshan-client-demo/src/main/java/org/eclipse/leshan/client/demo/LeshanClientDemo.java

```diff
@@ -872,4 +878,28 @@ public void run() {
             }
         }
     }
+
+    private static class ClientShutdownOnUnexpectedErrorObserver extends LwM2mClientObserverAdapter {
+        final LeshanClient client;
+        final MyDevice myDevice;
+        final RandomTemperatureSensor randomTemperatureSensor;
+
+        public ClientShutdownOnUnexpectedErrorObserver(
+            final LeshanClient client,
+			final MyDevice myDevice,
+            final RandomTemperatureSensor randomTemperatureSensor
+        ) {
+            this.client = client;
+            this.myDevice = myDevice;
+            this.randomTemperatureSensor = randomTemperatureSensor;
+        }
+
+        @Override
+        public void onUnexpectedErrorOccurred(Throwable unexpectedError) {
+            LOG.error("unexpected error occurred", unexpectedError);
+            client.destroy(true);
+            myDevice.destroy();
+            randomTemperatureSensor.destroy();
```

Comment:
I think we need to do the same `Lwm2mInstanceEnabler` should also implement (startable, stopable, destroyable)
And Object enabler should do pretty much the same as "LwM2mObjectTree" by interating over instance. (see https://github.com/eclipse/leshan/issues/933#issuecomment-732775930)

Reply:
Sounds good. I'll do this 👍

Reply:
fixed at 3f73020